tests/components/generic/test_config_flow.py

```diff
@@ -0,0 +1,510 @@
+"""Test The generic (IP Camera) config flow."""
+
+import errno
+from unittest.mock import patch
+
+import av
+import httpx
+import pytest
+import respx
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.generic.const import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_RTSP_TRANSPORT,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    DOMAIN,
+)
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+)
+
+from tests.common import MockConfigEntry
+
+TESTDATA = {
+    CONF_STILL_IMAGE_URL: "http://127.0.0.1/testurl/1",
+    CONF_STREAM_SOURCE: "http://127.0.0.1/testurl/2",
+    CONF_AUTHENTICATION: HTTP_BASIC_AUTHENTICATION,
+    CONF_USERNAME: "fred_flintstone",
+    CONF_PASSWORD: "bambam",
+    CONF_FRAMERATE: 5,
+    CONF_VERIFY_SSL: False,
+}
+
+TESTDATA_OPTIONS = {
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE: False,
+    **TESTDATA,
+}
+
+TESTDATA_YAML = {
+    CONF_NAME: "Yaml Defined Name",
+    **TESTDATA,
+}
+
+
+@respx.mock
+async def test_form(hass, fakeimg_png, mock_av_open, user_flow):
+    """Test the form with a normal set of settings."""
+
+    with mock_av_open as mock_setup:
+        result2 = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup function on all create entry results to avoid setting up the whole integration.

See the scaffold script test template code for example.

Reply:
Ok, done. 1c19dac